### PR TITLE
chore (refs DPLAN-11668): Add missing Hooks/Selectors in Public-View.

### PR DESCRIPTION
**Pull request by ahmad-demos**
### Ticket
https://demoseurope.youtrack.cloud/issue/DPLAN-11668/Hooks-Public-View-Hooks

- Add missing Hooks/Selectors in Public-View.

### CODE REVIEW

File: client/js/components/map/publicdetail/controls/MapTools.vue [hunk 1]
```diff
@@ -35,7 +36,8 @@
         :title="tool.title">
         <button
           :class="prefixClass('btn--blank o-link--default')"
-          :aria-label="tool.title + ' ' + Translator.trans('map.interactive.pointer.needed')">
+          :aria-label="tool.title + ' ' + Translator.trans('map.interactive.pointer.needed')"
+          :data-cy="`mapTools:${tool.title.replace(/\s+/g, '-').toLowerCase()}`">
```
Comment (salisdemos) [inline client/js/components/map/publicdetail/controls/MapTools.vue:40]:
```suggestion
          :data-cy="`mapTools:${tool.id}`">
```

File: client/js/components/map/publicdetail/controls/layerlist/DpPublicLayerListLayer.vue [hunk 2]
```diff
@@ -21,6 +21,7 @@
       <button
         :class="prefixClass('btn--blank btn--focus w-3 text-left')"
         :aria-label="layer.attributes.name + ' ' + statusAriaText"
+        :data-cy="`publicLayerListLayer:${layerTitle.replace(/\s+/g, '-').toLowerCase()}`"
```
Comment (salisdemos) [inline client/js/components/map/publicdetail/controls/layerlist/DpPublicLayerListLayer.vue:24]: do we have something more unique here? in the worst case maybe `id` ?

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: client/js/components/map/publicdetail/controls/layerlist/DpPublicLayerListLayer.vue (hunk 2 above)
Comment (ahmad-demos) [reply client/js/components/map/publicdetail/controls/layerlist/DpPublicLayerListLayer.vue:24]: When we use like `id`, then we have Hook like that `data-cy="publicLayerListLayer:1b4f6356-7222-11ec-92ee-b026282c0641"`

Comment (salisdemos) [reply client/js/components/map/publicdetail/controls/layerlist/DpPublicLayerListLayer.vue:24]: Thats why I asked, if there is something better ;-)

Comment (ahmad-demos) [reply client/js/components/map/publicdetail/controls/layerlist/DpPublicLayerListLayer.vue:24]: I will try to find something better :+1:

Comment (ahmad-demos) [reply client/js/components/map/publicdetail/controls/layerlist/DpPublicLayerListLayer.vue]: I found a solution || idea:
We have in DpPublicLayerList `layerType`  Prop, then we can make name with or belongs tolayerType!
For example:
Kartenebenen ein/ausblenden:
- Kreis-,Amts-,Gemeindegrenzen: `data-cy="publicLayerListLayer:overlay:0"`
Grundkarten:
- Digitale Orthophotos (Luftbilder): `data-cy="publicLayerListLayer:base:0"`
- Web-Atlas: `data-cy="publicLayerListLayer:base:1"`
- Digitale Topographische Karte: `data-cy="publicLayerListLayer:base:2"`

Done in https://github.com/demos-europe/demosplan-core/pull/3187/commits/d1e708bd125c6f9a1a94b55278f9d2b1eeb98097

**Review by salisdemos — APPROVED** [review]

**Pull request merged**